cmd/sidecar-initializer/main.go

```diff
@@ -35,68 +32,60 @@ func getRootCmd() *cobra.Command {
-		resyncPeriod time.Duration
+		port         int
 	}{}
 
 	rootCmd := &cobra.Command{
 		Use:   "sidecar-initializer",
 		Short: "Kubernetes initializer for Istio sidecar",
 		RunE: func(*cobra.Command, []string) error {
-			switch inject.InjectionPolicy(flags.policy) {
-			case inject.InjectionPolicyOff, inject.InjectionPolicyOptIn, inject.InjectionPolicyOptOut:
-			default:
-				return fmt.Errorf("unknown injection policy: %v", flags.policy)
-			}
-
-			client, err := kube.CreateInterface(flags.kubeconfig)
+			restConfig, client, err := kube.CreateInterface(flags.kubeconfig)
 			if err != nil {
 				return multierror.Prefix(err, "failed to connect to Kubernetes API.")
 			}
 
 			glog.V(2).Infof("version %s", version.Line())
-			glog.V(2).Infof("flags %s", spew.Sdump(flags))
 
-			// receive mesh configuration
-			mesh, err := cmd.ReadMeshConfig(flags.meshconfig)
+			config, err := inject.GetInitializerConfig(client, flags.namespace, flags.injectConfig)
 			if err != nil {
-				return multierror.Prefix(err, "failed to read mesh configuration.")
+				return multierror.Prefix(err, "failed to read initializer configuration")
 			}
 
-			options := inject.InitializerOptions{
-				ResyncPeriod: flags.resyncPeriod,
-				Hub:          flags.hub,
-				Tag:          flags.tag,
-				Namespace:    flags.namespace,
+			// retrieve mesh configuration separately
+			if config.Params.Mesh, err = cmd.ReadMeshConfig(flags.meshconfig); err != nil {
+				return multierror.Prefix(err, "failed to read mesh configuration.")
+			}
 
-				InjectionPolicy: inject.InjectionPolicy(flags.policy),
+			initializer, err := inject.NewInitializer(restConfig, config, client)
+			if err != nil {
+				return multierror.Prefix(err, "failed to create initializer")
 			}
-			initializer := inject.NewInitializer(client, mesh, options)
 
 			stop := make(chan struct{})
+
+			if flags.port != 0 {
+				server := inject.NewHTTPServer(flags.port, config)
+				go server.Run(stop)
+			}
 			go initializer.Run(stop)
-			cmd.WaitSignal(stop)
 
+			cmd.WaitSignal(stop)
 			return nil
 		},
 	}
 
-	rootCmd.PersistentFlags().StringVar(&flags.hub, "hub", "docker.io/istio", "Docker hub")
-	rootCmd.PersistentFlags().StringVar(&flags.tag, "tag", "0.2", "Docker tag")
-	rootCmd.PersistentFlags().StringVar(&flags.namespace, "namespace",
-		v1.NamespaceAll, "Namespace managed by initializer")
-	rootCmd.PersistentFlags().StringVar(&flags.policy, "policy",
-		string(inject.InjectionPolicyOff), "default injection policy")
-
 	rootCmd.PersistentFlags().StringVar(&flags.kubeconfig, "kubeconfig", "",
 		"Use a Kubernetes configuration file instead of in-cluster configuration")
 	rootCmd.PersistentFlags().StringVar(&flags.meshconfig, "meshconfig", "/etc/istio/config/mesh",
-		fmt.Sprintf("File name for Istio mesh configuration"))
-	rootCmd.PersistentFlags().DurationVar(&flags.resyncPeriod, "resync", 6*time.Minute,
-		"Initializers resync interval")
+		"File name for Istio mesh configuration")
```

Comment:
out of curiosity, why do we need to specify mesh config in sidecar initializer?

Reply:
`ProxyListenPort`, `AuthPolicy`, and `AuthCertsPath`. They're used in the sidecar template. We could dup these between inject and proxy-mesh ConfigMaps but that seemed more error prone at this stage than continuing to use use the Proxymesh directly.

Reply:
I prefer that we use a single source of truth for global settings, so this is sort of necessary. AuthPolicy will eventually become a destination policy. I also like that istioctl doesn't have to fetch it itself, but rather have this service do conversion.